packages/grafana-ui/src/components/Toggletip/Toggletip.tsx

```diff
@@ -0,0 +1,288 @@
+import { css } from '@emotion/css';
+import { Placement } from '@popperjs/core';
+import React, { useEffect, useRef } from 'react';
+import { usePopperTooltip } from 'react-popper-tooltip';
+
+import { colorManipulator, GrafanaTheme2 } from '@grafana/data';
+
+import { useStyles2 } from '../../themes/ThemeContext';
+import { IconButton } from '../IconButton/IconButton';
+import { Portal } from '../Portal/Portal';
+
+import { ToggletipContent } from './types';
+
+export interface ToggletipProps {
+  /** The theme used to display the toggletip */
+  theme?: 'info' | 'error';
+  /** The title to be displayed on the header */
+  title?: JSX.Element | string;
+  /** determine whether to show or not the close button **/
+  closeButton?: boolean;
+  /** Callback function to be called when the toggletip is closed */
+  onClose?: Function;
+  /** The preferred placement of the toggletip */
+  placement?: Placement;
+  /** The text or component that houses the content of the toggleltip */
+  content: ToggletipContent;
+  /** The text or component to be displayed on the toggletip's bottom */
+  footer?: JSX.Element | string;
+  /** The UI control users interact with to display toggletips */
+  children: JSX.Element;
+}
+
+export const Toggletip = React.memo(
+  ({
+    children,
+    theme = 'info',
+    placement = 'auto',
+    content,
+    title,
+    closeButton = true,
+    onClose,
+    footer,
+  }: ToggletipProps) => {
+    const styles = useStyles2(getStyles);
+    const style = styles[theme];
+    const contentRef = useRef(null);
+    const [controlledVisible, setControlledVisible] = React.useState(false);
+
+    const closeToggletip = () => {
+      setControlledVisible(false);
+      onClose?.();
+    };
+
+    useEffect(() => {
+      if (controlledVisible) {
+        const handleKeyDown = (enterKey: KeyboardEvent) => {
+          if (enterKey.key === 'Escape') {
+            closeToggletip();
+          }
+        };
+        document.addEventListener('keydown', handleKeyDown);
+        return () => {
+          document.removeEventListener('keydown', handleKeyDown);
+        };
+      }
+      return;
+    }, [controlledVisible]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Would be better if we can avoid disabling the lint, we can wrap the `closeToggletip` in a `useCallback` and then use it here as a dependency?